src/helpers/paywall-variables-helpers.ts

```diff
@@ -148,87 +123,91 @@ function getTotalPriceAndPerMonth({
   price,
   period,
   full = false,
-  selectedLocale,
+  translator,
 }: {
   price: Price;
   period: Period | null;
   full?: boolean;
-  selectedLocale: string;
+  translator: Translator;
 }) {
   if (!period || !period.number) return price.formattedPrice;
-  let pricePerMonth = "";
-  const monthPeriodString = full ? "month" : "mo";
-  const periodString = full ? period.unit : AbbreviatedPeriod[period.unit];
+  let pricePerMonth: string | undefined = "";
+
   if (period.unit === "year") {
-    pricePerMonth = formatPrice(
+    pricePerMonth = translator.formatPrice(
       price.amountMicros / 12,
       price.currency,
-      selectedLocale,
     );
-    return `${price.formattedPrice}/${period.number}${periodString}($${pricePerMonth}/(${monthPeriodString})`;
   }
+
   if (period.unit === "month") {
-    pricePerMonth = formatPrice(
+    pricePerMonth = translator.formatPrice(
       price.amountMicros / period.number,
       price.currency,
-      selectedLocale,
     );
-    if (period.number > 1) {
-      return `${price.formattedPrice}/${period.number}${periodString}(${pricePerMonth}/${monthPeriodString})`;
-    } else {
-      return `${price.formattedPrice}`;
+
+    if (period.number === 1) {
+      return price.formattedPrice;
```

Comment:
should we move this a bit above as an early return pre calling translator?

Reply:
Done!